Arrangement_on_surface_2/include/CGAL/Arrangement_on_surface_2.h

```diff
@@ -1516,6 +1527,37 @@ class Arrangement_on_surface_2 {
 
   //@}
 
+  /*!
+   * Cleans the inner CCB if sweep mode was used, by removing all
+   * non-valid inner CCBs
+   */
+  void clean_inner_ccbs()
+  {
+    for (DHalfedge_iter he = _dcel().halfedges_begin();
+         he != _dcel().halfedges_end(); ++ he)
+    {
+      if (!he->is_on_inner_ccb())
+        continue;
+
+      DInner_ccb* ic1 = he->inner_ccb_no_redirect();
+      if (ic1->is_valid())
+        continue;
+
+      DInner_ccb* ic2 = he->inner_ccb();
+      if (!ic2->halfedge()->is_on_inner_ccb()
+          || ic2->halfedge()->inner_ccb_no_redirect() != ic2)
+        ic2->set_halfedge(&(*he));
```

Comment:
I'm a bit surprised that this is needed.

Reply:
It's actually not, it's remaining from an older version of the code which needed it. I just tested and it never happens (as expected). I'll replace it with an assertion.